### ISSUE TITLE
`var`:ness is lost with distinct and generics

### ISSUE THREAD

**Issue by arnetheduck** [nim-lang/Nim#21123]
### Description

```nim
import std/typetraits
  
proc a(v: var seq[int]) = echo v


type AAA[X] = distinct seq[int]
type BBB = distinct AAA[bool]

var bbb: BBB

a(distinctBase distinctBase(bbb))
```

```
/usercode/in.nim(11, 2) Error: type mismatch: got <seq[int]>
but expected one of:
proc a(v: var seq[int])
  first type mismatch at position: 1
  required type for v: var seq[int]
  but expression '
[type node](bbb)' is immutable, not 'var'

expression: a do:
  [type node](bbb)
```

https://play.nim-lang.org/#ix=4iMi

### Nim Version

1.6.10, devel

**Comment by bung87** [comment]
it compiles and runs fine in 100b304ac2aaa65222d7a9043acd0c534ed484cb

**Comment by arnetheduck** [comment]
@bung87 do you know what fixed it? it doesn't run on play.nim-lang.org devel

**Comment by bung87** [comment]
don't know which fixed it,I just run it locally

**Comment by arnetheduck** [comment]
@narimiran possible to find fix and backport to 1.6?

**Comment by bung87** [comment]
https://github.com/nim-lang/Nim/pull/18837  might be this one, un verified.
